packages/jellyfish-api-core/__tests__/category/loan/auctionBid.test.ts

```diff
@@ -0,0 +1,412 @@
+}
+
+beforeEach(async () => {
+  await tGroup.start()
+  await setup()
+})
+
+afterEach(async () => {
+  await tGroup.stop()
+})
+
+describe('test auctionBid', () => {
+  it('should auctionBid', async () => {
+    const auctionsBefore = await bob.container.call('listauctions')
+    expect(auctionsBefore).toStrictEqual([
+      {
+        vaultId: bobVaultId,
+        batchCount: 2,
+        liquidationPenalty: 5,
+        batches: [
+          {
+            index: 0,
+            collaterals: ['5000.00000000@DFI', '0.50000000@BTC'],
+            loan: '500.00000000@TSLA'
+          },
+          {
+            index: 1,
+            collaterals: ['5000.00000000@DFI', '0.50000000@BTC'],
+            loan: '500.00000000@TSLA'
+          }
+        ]
+      }
+    ])
+
+    const bobColAccBefore = await bob.rpc.account.getAccount(bobColAddr)
+    expect(bobColAccBefore).toStrictEqual(['8900.00000000@DFI', '545.45454546@TSLA'])
+
+    const bobTSLAAccBefore = bobColAccBefore.length > 0
+      ? bobColAccBefore.find((amt: string) => amt.split('@')[1] === 'TSLA')
+      : undefined
+    const bobTSLAAmtBefore = bobTSLAAccBefore !== undefined ? Number(bobTSLAAccBefore.split('@')[0]) : 0
+
+    const txid = await bob.rpc.loan.auctionBid({
+      vaultId: bobVaultId,
+      index: 0,
+      from: bobColAddr,
+      amount: '525@TSLA' // (500 * 5%) + 500 = 525, min first bid is 525
+    })
+    expect(typeof txid).toStrictEqual('string')
+    await bob.container.generate(1)
+    await tGroup.waitForSync()
+
+    const bobColAccAfter = await bob.rpc.account.getAccount(bobColAddr)
+    expect(bobColAccAfter).toStrictEqual(['8900.00000000@DFI', '20.45454546@TSLA'])
+
+    const bobTSLAAccAfter = bobColAccAfter.length > 0
+      ? bobColAccAfter.find((amt: string) => amt.split('@')[1] === 'TSLA')
+      : undefined
+    const bobTSLAAmtAfter = bobTSLAAccAfter !== undefined ? Number(bobTSLAAccAfter.split('@')[0]) : 0
+
+    expect(bobTSLAAmtBefore - bobTSLAAmtAfter).toStrictEqual(525) // 545.45454546 - 20.45454546 = 525
+
+    const aliceColAccBefore = await alice.rpc.account.getAccount(aliceColAddr)
+    expect(aliceColAccBefore).toStrictEqual(['30000.00000000@DFI', '29999.00000000@BTC', '10000.00000000@TSLA'])
+
+    // test second round auctionBid
+    await alice.rpc.loan.auctionBid({
+      vaultId: bobVaultId,
+      index: 0,
+      from: aliceColAddr,
+      amount: '535@TSLA' // (525 * 1%) + 525 = 530.25
+    })
+    await alice.generate(1)
+    await tGroup.waitForSync()
+
+    // end the auction and alice win the bid
+    await bob.generate(36)
+
+    const auctionsAfter = await bob.container.call('listauctions')
+    expect(auctionsAfter).toStrictEqual([
+      {
+        vaultId: bobVaultId,
+        batchCount: 1,
+        liquidationPenalty: 5,
+        batches: [
+          {
+            index: 0,
+            collaterals: ['5000.00000000@DFI', '0.50000000@BTC'],
+            loan: '509.52380952@TSLA'
```

Comment:
how we get this number?

Reply:
hmm... yea... i need double check it
from this line??
https://github.com/DeFiCh/ain/blob/73a8a134ce730cc61dddc298f8eeeb498099c747/src/validation.cpp#L3090-L3091